Tests/SonarQube.MSBuild.Tasks.UnitTests/WriteProjectInfoFileTests.cs

```diff
@@ -220,20 +222,415 @@ public void WriteProjectInfoFile_UnrecognisedLanguages()
+            Assert.AreEqual(actualVBNet.Encoding, null, "unexpected encoding");
+        }
+
+        [TestMethod]
+        public void WriteProjectInfoFile_Execute_WhenNotCSharpWithNullCodePage_ExpectsNull()
+        {
+            // Arrange
+            var encodingProvider = new TestEncodingProvider(
+                x =>
+                {
+                    Assert.Fail("should not have been called");
+                    throw new InvalidOperationException();
+                },
+                x =>
+                {
+                    Assert.Fail("should not have been called");
+                    throw new InvalidOperationException();
+                });
+
+            // Act
+            var actual = WriteProjectInfoFile_ExecuteAndReturn(encodingProvider, null, "whatever", "foo9");
+
+            // Assert
+            Assert.AreEqual(actual.Encoding, null, "unexpected encoding");
+        }
+
+        [TestMethod]
+        public void WriteProjectInfoFile_Execute_WhenNotCSharpVBNetWithInferiorToZeroCodePage_ExpectsNull()
+        {
+            // Arrange
+            var encodingProvider = new TestEncodingProvider(
+                x =>
+                {
+                    Assert.Fail("should not have been called");
+                    throw new InvalidOperationException();
+                },
+                x =>
+                {
+                    Assert.Fail("should not have been called");
+                    throw new InvalidOperationException();
+                });
+
+            // Act
+            var actual = WriteProjectInfoFile_ExecuteAndReturn(encodingProvider, "-1", "whatever", "foo10");
+
+            // Assert
+            Assert.AreEqual(actual.Encoding, null, "unexpected encoding");
+        }
+
+        [TestMethod]
+        public void WriteProjectInfoFile_Execute_WhenNotCSharpVBNetWithBiggerThanLongCodePage_ExpectsNull()
+        {
+            // Arrange
+            var encodingProvider = new TestEncodingProvider(
+                x =>
+                {
+                    Assert.Fail("should not have been called");
+                    throw new InvalidOperationException();
+                },
+                x =>
+                {
+                    Assert.Fail("should not have been called");
+                    throw new InvalidOperationException();
+                });
+
+            var codePage = ((double)long.MaxValue + 1).ToString();
+
+            // Act
+            var actual = WriteProjectInfoFile_ExecuteAndReturn(encodingProvider, codePage, "whatever", "foo11");
+
+            // Assert
+            Assert.AreEqual(actual.Encoding, null, "unexpected encoding");
+        }
+
+        [TestMethod]
+        public void WriteProjectInfoFile_Execute_WhenNotCSharpVBNetWithNonExistingCodePage_ExpectsNull()
+        {
+            // Arrange
+            var encodingProvider = new TestEncodingProvider(
+                x =>
+                {
+                    if (x == 42)
```

Comment:
I really don't like `42` and `"42"` showing up in this method. There should be one single variable.